plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -140,20 +138,53 @@ public ConnectorTableMetadata getTableMetadata()
         return connectorTableMetadata;
     }
 
-    private Optional<RowType> getPartitionColumnType(List<PartitionField> fields, Schema schema)
+    private static List<PartitionField> getAllPartitionFields(Table icebergTable)
     {
+        Set<Integer> existingColumnsIds = icebergTable.schema()
+                .columns().stream()
+                .map(NestedField::fieldId)
+                .collect(toUnmodifiableSet());
+
+        List<PartitionField> visiblePartitionFields = icebergTable.specs()
+                .values().stream()
+                .flatMap(partitionSpec -> partitionSpec.fields().stream())
+                // skip columns that were dropped
+                .filter(partitionField -> existingColumnsIds.contains(partitionField.sourceId()))
+                .collect(toImmutableList());
+
+        return filterOutDuplicates(visiblePartitionFields);
```

Comment:
Use `Stream#distinct`?

Reply:
I can't rely on `PartitionField` implementation of `hashCode` and `equals` as they take transformation into account and I really care about Id's. I could maybe provide my own comparator or something but I think this is cleaner now